AMM/project/scripts/swap-exact-input/src/main.sw

```diff
@@ -0,0 +1,60 @@
+script;
+
+use libraries::{AMM, Exchange};
+
+enum InputError {
+    RouteTooShort: (),
+}
+
+enum SwapError {
+    ExcessiveSlippage: u64,
+    PairExchangeNotRegistered: (ContractId, ContractId),
+}
+
+fn main(
+    amm_contract_address: ContractId,
+    assets: Vec<ContractId>,
+    input_amount: u64,
+    minimum_output_amount: Option<u64>,
+    deadline: u64,
+) -> u64 {
+    require(assets.len() >= 2, InputError::RouteTooShort);
+
+    let amm_contract = abi(AMM, amm_contract_address.into());
+
+    let mut latest_bought = input_amount;
+    let intermediate_minimum_amount: Option<u64> = Option::None;
```

Comment:
This variable is never changed, as we only check the final amount out. Suggest removing and just passing `Option::None` directly to the swap call.

Reply:
Right.